src/components/Header/index.js

```diff
@@ -1,18 +1,54 @@
 const React = require('react');
+const T = require('prop-types');
 const NavLink = require('react-router-dom').NavLink;
 const Classes = require('./styles.scss');
 
-module.exports = () => (
-
-    <div>
-        <h1>Strangeluv</h1>
-        <NavLink exact to='/' activeClassName={Classes.activeRoute}>
-            Home
-        </NavLink>
-        {' · '}
-        <NavLink to='/counter' activeClassName={Classes.activeRoute}>
-            Counter
-        </NavLink>
-    </div>
-
-);
+module.exports = class Header extends React.Component {
+
+    static propTypes = {
+        logout: T.func.isRequired,
+        isAuthenticated: T.bool.isRequired
+    };
+
+    renderNotAuthenticated = () => {
```

Comment:
Are we using this syntax now? ¯\_(ツ)_/¯

Reply:
For anybody wondering what this syntax does, it sets a func on the class that automatically binds `this` to be the class

- Replacing the need for `this.myFunc = this._myFunc.bind(this)` in the constructor

Reply:
I personally prefer arrow functions because it's tidier and less repetitive; what's the advantage of binding in the constructor? Performance-wise it's about the same, correct?

Reply:
I agree that it's more concise!  And I'm not worried about performance.  My issues with it are that

 - it's not yet standard javascript, and I'm not sure if it will be.
 - it's really weird because in all other situations an "arrow function" specifically does _not_ have its own `this`!  This is really confusing.
 - it's more explicit to bind for the purposes passing a method to props.  I think we sometimes forget _why_ the binding is necessary in the first place.
 - there are other options even _aside_ from binding—it's no longer against our lint to use arrow functions in `render()`, for example.  So we could do `prop={() => this.myMethod()}` in many cases, and author our methods in the typical (non-arrow) way.

Reply:
how do you want me to author methods, bind in constructor? i'll update everywhere, just gimme the word!

Reply:
Until we collectively land on something else, I would suggest binding them in the constructor as we do in other projects.  (No need to bind methods that aren't passed directly as props to a component.)  We've  been exposed to different projects, though, so maybe I'm wrong in thinking that we don't do this elsewhere.  But maybe we should talk about this in strangeluv standup!

Reply:
@jdonisvitch snuck it into cupie some, but that's the only one I'm aware of.